chrome-extension/popup.js

```diff
@@ -0,0 +1,90 @@
+const API_KEY = 'AIzaSyAed5yuxCNrXlEFAqSoWwAQpE3Ng95Tzl8';
+const DISCOVERY_DOCS = [
+  "https://docs.googleapis.com/$discovery/rest?version=v1",
+  "https://www.googleapis.com/discovery/v1/apis/drive/v3/rest"
+];
+
+const NOTES_TEMPLATE_DOC_ID = '1XlcAy-vrleXBxJl5Qy_SxGUyTqcwdUIhyJI2BygpNEc';
+const REVOKE_TOKEN_URL = 'https://accounts.google.com/o/oauth2/revoke?token=';
+
+let loggedIn = false;
+
+/**
+ * Initializes Google APIs client library for browser-side JavaScript (gapi)
+ * and the button functions when script is loaded.
+ */
+function onGAPILoad() {
+  document.getElementById('generate-note-btn').onclick = generateNote;
+  document.getElementById('logout-btn').onclick = logout;
+  document.getElementById('doc-name-input').value = 'gNote ' + getDate();
+
+  gapi.client.init({
+    apiKey: API_KEY,
+    discoveryDocs: DISCOVERY_DOCS,
+  }).then(function() {
+    chrome.identity.getAuthToken({interactive: true}, function(token) {
+      if (chrome.runtime.lastError) {
+        return;
+      }
+      gapi.auth.setToken({
+        access_token: token,
+      });
+      loggedIn = true;
+      setAccountInfo();
+    })
+  }, function(error) {
+    console.log('error', error)
+  });
+}
+
+/** Returns formatted string of today's date. */
+function getDate() {
+  const today = new Date();
+  const dd = String(today.getDate()).padStart(2, '0');
+  const mm = String(today.getMonth() + 1).padStart(2, '0');
+  const yyyy = today.getFullYear();
+
+  return mm + '/' + dd + '/' + yyyy;
+}
+
+/**
+ * Generates a copy of the notes template and opens a new tab
+ * with the newly created Google Doc.
+ */
+function generateNote() {
+  // TODO: Add generate note functionality
+}
+
+/**
+ * Logs out the user by revoking their authentication token and removing
+ * it from the cache.
+ */
+function logout() {
+  chrome.identity.getAuthToken({interactive: false}, function(token) {
+    var url = REVOKE_TOKEN_URL + token;
```

Comment:
nit: change var to const
```suggestion
    const url = REVOKE_TOKEN_URL + token;
```

Reply:
Oops I missed that, I'll change it!